pyvista/core/filters.py

```diff
@@ -2189,6 +2192,51 @@ def compute_gradient(dataset, scalars=None, gradient_name='gradient',
         gradient_name : str, optional
             The name of the output array of the computed gradient.
 
+        preference: str, optional
+            Data type preference. Either 'point' or 'cell'.
+
+        """
+        logging.warning('DEPRECATED: ``.compute_gradient`` is deprecated. Use ``.compute_derivative`` instead.')
```

Comment:
Should we also use `warnings.DeprecationWarning()` here as well? I'm not sure where the logging reports it's errors, as I've only seen it implementing as a file logging utility.

Reply:
Logging will print these warnings to stdout usually. I've experienced problems with `warnings` not showing up in the right place (or at all when using `DeprecationWarning`). AFIK, we can count on `logging.warning` doing the job and we can easily filter those logs in downstream projects to catch these deprecations/issues.

It may be worth refactoring this deprecation warning into its own class so that we can easily/consistently implemnt this everywhere. Any changes to this should come in a separate PR

Reply:
For this PR, that notice is perfect